File: src/main/java/com/emarket/service/interfaz/CarritoService.java
package com.emarket.service.interfaz;

import com.emarket.dto.carrito.ActualizarCantidadCarritoRequestDto;
import com.emarket.dto.carrito.AgregarItemCarritoRequestDto;
import com.emarket.dto.carrito.CarritoResponseDto;
import com.emarket.dto.pedido.PedidoResponseDto;

public interface CarritoService {
    CarritoResponseDto verCarrito(String token);
    CarritoResponseDto agregarItem(String token, AgregarItemCarritoRequestDto dto);
    CarritoResponseDto actualizarCantidad(String token, Long itemId, ActualizarCantidadCarritoRequestDto dto);
    CarritoResponseDto eliminarItem(String token, Long itemId);
    PedidoResponseDto confirmarCompra(String token);
}

File: frontend/src/services/carrito.service.ts
import { apiClient } from '../core/api/axiosInstance';
import type {
  AgregarItemCarritoRequestDto,
  ActualizarCantidadCarritoRequestDto,
  CarritoResponseDto,
  ConfirmarCompraRequestDto,
  ApiResponseDto,
  PedidoResponseDto,
} from '../core/types';

export const carritoService = {
  async ver(): Promise<CarritoResponseDto> {
    const { data } = await apiClient.get<CarritoResponseDto>('/carrito');
    return data;
  },

  async agregarItem(dto: AgregarItemCarritoRequestDto): Promise<CarritoResponseDto> {
    const { data } = await apiClient.post<CarritoResponseDto>('/carrito/items', dto);
    return data;
  },

  async actualizarCantidad(
    itemId: number,
    dto: ActualizarCantidadCarritoRequestDto,
  ): Promise<CarritoResponseDto> {
    const { data } = await apiClient.put<CarritoResponseDto>(`/carrito/items/${itemId}`, dto);
    return data;
  },

  async eliminarItem(itemId: number): Promise<ApiResponseDto<CarritoResponseDto>> {
    const { data } = await apiClient.delete<ApiResponseDto<CarritoResponseDto>>(
      `/carrito/items/${itemId}`,
    );
    return data;
  },

  async confirmarCompra(
    dto: ConfirmarCompraRequestDto,
  ): Promise<ApiResponseDto<PedidoResponseDto>> {
    const { data } = await apiClient.post<ApiResponseDto<PedidoResponseDto>>(
      '/carrito/confirmar',
      dto,
    );
    return data;
  },
};
